analytics/analytics-ui/src/app/services/filter.pipe.ts

```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
+@Pipe({
+    name: 'filter'
+})
+export class FilterPipe implements PipeTransform {
+    transform(items: any[], searchText: string): any[] {
```

Comment:
It seems dangerous to use any[] here. You should make use of typescript and create an Analytic class. Also, you should consider renaming this to 'analytic-filter' as it's specific to analytics.